### PR TITLE
feat: Cache request from Identity popover

### LINKED ISSUE #2505: Cache request from Identity popover
### Is your feature request related to a problem?

I just realized that spotlight does more than **140** queries to magick, a vast number.

### Describe the solution you would like

I would like to see that these requests are cached for 12hrs or so

### Describe alternatives you have considered

Use Vuex :)

### Screeenshots

Screenshot 2022-03-05 at 17 47 58

Screenshot 2022-03-05 at 17 48 20

### PR DESCRIPTION
**Pull request by zhengow**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2505
- [x] Cache request from Identity popover

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </spotlight>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

https://user-images.githubusercontent.com/38389586/157069185-549e427e-9cb6-47a3-91cd-fab37fa7bb1f.mp4

### CODE REVIEW

**Comment by zhengow** [comment]
> As would [Mr Dyatlov](https://tvtropes.org/pmwiki/pmwiki.php/Quotes/Chernobyl) say: `3.6 - not great, not terrible.`
> 
> The algo:
> 
> * get the data from the storage (vuex, or IDB)
> * check if the difference between `updatedAt` and `Date.now()` is more than 12hrs -- if not than show the data
> * if yes fetch the data from the graphql
> * show the data
> * add to the storage
> 
> Where is the issue? You are magically labeling something as type `cache` and then it's never updated.
> 
> Add 2. **Try to unwrap the data which you save into the storage**
> 
> How it looks now:
> 
> ```ts
> export interface MintInfo {
>    totalCreated: number
>    totalCollected: number
>    totalSold: number
>    firstMintDate: Date
>  }
> ```
> 
> ```ts
> export interface MintInfo {
>    firstMint: { nodes: any[] }
>    nFTCollected: { totalCount: number }
>    nFTCreated: { totalCount: number }
>    nFTSold: { totalCount: number }
>  }
> ```

already change, plz check:)

File: components/shared/IdentityPopover.vue [hunk 1]
```diff
@@ -103,39 +104,70 @@ export default class IdentityPopover extends mixins(
...
+    data: MintInfo | any
+    type?: 'cache'
+  }) {
+    if (type === 'cache') {
+      this.totalCreated = data.totalCreated
+      this.totalCollected = data.totalCollected
+      this.totalSold = data.totalSold
+      this.firstMintDate = data.firstMintDate
+    } else {
+      if (data) {
```
Comment (vikiival) [inline components/shared/IdentityPopover.vue:152]:
```suggestion
    } else if (data) {
```

File: components/shared/IdentityPopover.vue [hunk 2]
```diff
@@ -103,39 +104,70 @@ export default class IdentityPopover extends mixins(
...
-        manual: true,
-        client: this.urlPrefix,
-        loadingKey: 'isLoading',
-        result: this.handleResult,
-        variables: () => {
-          return {
+      const data = this.$store.getters['identityMint/getIdentityMintFor'](
+        this.identity.address
+      )
+      if (data && Date.now() - data.updatedAt < 12 * 60 * 60 * 1000) {
```
Comment (vikiival) [inline components/shared/IdentityPopover.vue:113]: from date fns

```suggestion
      if (isBefore(data.updatedAt, subHours(Date.now, 12))) {
```

**Review by vikiival — CHANGES_REQUESTED** [review]
Looks really good mate! 
Just 💅 polish a couple of thigs and we are good to go 🚀

File: components/shared/IdentityPopover.vue (hunk 2 above)
Comment (zhengow) [reply components/shared/IdentityPopover.vue]: Already change, u really know much🙂
`const data = this.$store.getters['identityMint/getIdentityMintFor'](
        this.identity.address
      )
      if (
        data?.updatedAt &&
        isAfter(data.updatedAt, subHours(Date.now(), 12))
      ) {
        // if cache exist and within 12h
        await this.handleResult({ data, type: 'cache' })
      } else {
        const query =
          this.urlPrefix === 'rmrk'
            ? await import('@/queries/nftStatsByIssuer.graphql')
            : await import('@/queries/unique/nftStatsByIssuer.graphql')`

Comment (zhengow) [reply components/shared/IdentityPopover.vue]: > Already change, u really know much🙂 
` if ( data?.updatedAt && isAfter(data.updatedAt, subHours(Date.now(), 12)) ) 
{ // if cache exist and within 12h await this.handleResult({ data, type: 'cache' }) } 
else { ...
`

**Review by vikiival — APPROVED** [review]
did NOT test, but code looks OKI

**Comment by yangwao** [comment]
seem does the job, pre-cached and seventh page i haven't visited 


https://user-images.githubusercontent.com/5887929/158579931-7c7090e8-4ecc-4a6a-bba1-1837e227c036.mov

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 126.82 USD/KSM ~ 0.789 $KSM
🧗 [EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws](https://kusama.subscan.io/account/EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)
🔗 [0x8976dc16a4b4a52cd103fe41d36d14c963f42d36814a1c6c22ab0b84ef30f9ca](https://kusama.subscan.io/extrinsic/0x8976dc16a4b4a52cd103fe41d36d14c963f42d36814a1c6c22ab0b84ef30f9ca)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**